### DISCUSSION TITLE
How long will the recorded interface information be kept !

### DISCUSSION THREAD

**Discussion by Pactortester** [mitmproxy/mitmproxy#5290] [Q&A] [answered]
- In the mitmweb page, how long will the recorded interface information be kept? without the server shutting down.

**Comment by mhils** [comment, marked as answer]
Mitmproxy and mitmweb keep all flows in memory indefinitely (until the server is shut down or the flow list is cleared).

**Reply by Pactortester** [reply]
> Mitmproxy and mitmweb keep all flows in memory indefinitely (until the server is shut down or the flow list is cleared).

When the data volume of flows is relatively large, all of them are stored in memory. Will it affect the performance over time?

**Reply by mhils** [reply]
Yes. If you want long-running operation, you should look at mitmdump.
